iib/workers/tasks/build_merge_index_image.py

```diff
@@ -285,3 +287,63 @@ def handle_merge_request(
     set_request_state(
         request_id, 'complete', 'The index image was successfully cleaned and updated.'
     )
+
+
+def filter_invalid_bundles(bundle_paths, valid_ocp_version):
+    """
+    Filter bundles which are invalid for the index ocp_version.
+
+    :param list bundle_paths: list of bundles to be validated against the index ocp version.
+    :param str valid_ocp_version: the index ocp version against which the bundles will be validated.
+    :return: a list of valid bundles which satisfy the index ocp_version
+    :rtype: list
+    :raises IIBError: if bundles have a semantically invalid version set.
+
+           |  "v4.5"   |   "=v4.6"    | "v4.5-v4.7" | "v4.5,v4.6"
+    ---------------------------------------------------------------
+    v4.5   | included  | NOT included |  included   |  included
+    ---------------------------------------------------------------
+    v4.6   | included  |   included   |  included   |  included
+    ---------------------------------------------------------------
+    v4.7   | included  | NOT included |  included   |  included
+    """
+    index_valid_bundles = []
+    invalid_version_bundles = {}
+    float_valid_ocp_version = float(valid_ocp_version.replace('v', ''))
+    for bundle in bundle_paths:
+        try:
+            bundle_version_label = get_image_label(bundle, 'com.redhat.openshift.versions')
+            bundle_version = bundle_version_label.replace('v', '')
+            if bundle_version_label.startswith('='):
+                bundle_version = float(bundle_version.strip('='))
+                if bundle_version == float_valid_ocp_version:
+                    index_valid_bundles.append(bundle)
```

Comment:
the `bundle_version` not being the same as the `float_valid_ocp_version` would not exactly mean that the bundle version is invalid. It would simply mean that it's not valid for this particular target_index. It might be valid in general for some other ocp_version